src/drive/web/modules/views/Folder/syncHelpers.js

```diff
@@ -0,0 +1,146 @@
+/**
+ * Whether there is a file referenced by a share id
+ * @param {array} queryResults - List of folders and files
+ * @param {string} sharingId - Id of an io.cozy.sharings doc
+ * @returns {bool} true|false
+ */
+export const isThereFileReferencedBySharingId = (queryResults, sharingId) => {
+  return queryResults.some(query => {
+    return query.data.some(file => {
+      const fileReferences =
+        file.referenced_by && file.referenced_by.length >= 1
+      if (fileReferences) {
+        return file.referenced_by.some(reference => {
+          if (reference.type === 'io.cozy.sharings') {
+            return reference.id === sharingId
+          }
+          return false
+        })
+      }
+      return false
+    })
+  })
+}
+
+/**
+ * Remove a share from sharing context
+ * @param {object} params - Params
+ * @param {object} params.sharingsValue - Sharing Context value
+ * @param {function} params.setSharingsValue - Sharing Context setter
+ * @param {string} params.sharingId - Id of an io.cozy.sharings doc
+ */
+export const removeSharingFromContext = ({
+  sharingsValue,
+  setSharingsValue,
+  sharingId
+}) => {
+  delete sharingsValue[sharingId]
+  setSharingsValue(sharingsValue)
+}
+
+/**
+ * Create a syncing fake file
```

Comment:
We should also explain when this is needed. Because reading from the method's name I can understand that it creates à syncing fake file, but I'm interested in why / when